### PR TITLE
[IGA] Output Quadrature Domain Process

### PR DESCRIPTION
**Pull request by tteschemacher**
**Description**
Adds the possibility to write the locations of the integration points for an eased post processing.

### CODE REVIEW

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor
+    OutputQuadratureDomainProcess(
+        Model& rModel,
+        Parameters ThisParameters);
+
+    /// Destructor.
+    virtual ~OutputQuadratureDomainProcess() {}
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: virtual ~OutputQuadratureDomainProcess() = default

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Called only once befor the solution loop
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: typo

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 3]
```diff
@@ -17,6 +17,8 @@
 #include "includes/define.h"
 #include "custom_python/add_custom_utilities_to_python.h"
 
+#include "custom_processes/output_quadrature_domain_process.h"
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp]: I would add this to a new file called add_custom_processes_to_python and add this there. It is done like this in most of the other apps.

Comment (tteschemacher) [reply applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp]: as you wish haha ;)

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.h (hunk 2 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: okidoki

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.h (hunk 1 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: yup

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.h [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class OutputQuadratureDomainProcess
+ * @ingroup KratosCore
+ * @brief This class is used in order to check results using a json file containing the solution a given model part with a certain frequency
+ * @details This stores the dababase in a class denominated ResultDatabase which considers Table to store the information, therefore being able to interpolate results
+ * @author Vicente Mataix Ferrandiz
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: Shouldn't this be your name? Furthermore, I don't completely understand the description.

Comment (dagmawibekel) [reply applications/IgaApplication/custom_processes/output_quadrature_domain_process.h]: And my final comment is that you should check Codacy.

**Comment by tteschemacher** [comment]
@dagmawibekel are you fine with merging this one?

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp [hunk 5]
```diff
@@ -44,20 +44,20 @@ void OutputQuadratureDomainProcess::ExecuteBeforeSolutionLoop()
     if (OutputGeometryElements) {
         for (auto element : r_model_part.Elements()) {
             auto integration_point = element.GetGeometry().IntegrationPoints()[0];
-            contents += '[' + element.Id() + ',' + element.GetGeometry().GetGeometryParent(0).Id() + ',[';
+            contents += '[' + element.Id() + ',' + element.GetGeometry().GetGeometryParent(0).Id() + ",[";
```
Comment (rickyaristio) [inline applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp]: It should be std::to_string(element.Id()) and std::to_string(element.GetGeometry().GetGeometryParent(0).Id()).

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp [hunk 6]
```diff
@@ -44,20 +44,20 @@ void OutputQuadratureDomainProcess::ExecuteBeforeSolutionLoop()
...
             contents += std::to_string(integration_point[0]) + ',' + std::to_string(integration_point[1]) + "]],";
         }
     }
     if (OutputGeometryConditions) {
         for (auto condition : r_model_part.Conditions()) {
             auto integration_point = condition.GetGeometry().IntegrationPoints()[0];
-            contents += '[' << condition.Id() + ',' + condition.GetGeometry().GetGeometryParent(0).Id() + ',[';
+            contents += '[' + condition.Id() + ',' + condition.GetGeometry().GetGeometryParent(0).Id() + ",[";
```
Comment (rickyaristio) [inline applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp]: same

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp (hunk 5 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp]: thanks for the comment. I updated it.

File: applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp (hunk 6 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_processes/output_quadrature_domain_process.cpp]: cheers, I fixed it.

**Comment by tteschemacher** [comment]
ping @dagmawibekel 

@rickyaristio and I are using this process already successfully to post process in Rhino.

**Review by dagmawibekel — APPROVED** [review]
Awesome!

**Comment by tteschemacher** [comment]
Sweet

**Pull request merged**